app/code/Magento/Variable/Model/Variable.php

```diff
@@ -162,4 +179,39 @@ public function getVariablesOptionArray($withGroup = false)
         }
         return $variables;
     }
+
+    /**
+     * Check if the given store (or the current one) has the given
+     * variable set with the given text value
+     *
+     * @param string $variableCode
+     * @param string $expectedValue
+     * @param string $type
+     * @param int $storeId
+     *
+     * @return boolean
+     */
+    public function customVariableHasValue(
+        $variableCode,
+        $expectedValue,
+        $type = self::TYPE_TEXT,
+        $storeId = null
+    ) {
+        $status = false;
+
+        if ($storeId === null) {
+            $storeId = $this->storeManager->getStore()->getId();
+        }
+
+        $this->setStoreId($storeId)
+            ->loadByCode($variableCode);
```

Comment:
Since `EntityManager` introduction I don't think classes may be responsible for both entity and entity loading (not sure why `loadByCode` is still not deprecated, it should be according to my understanding).